### PR TITLE
imdsv2-configuration

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/3403

- Add an option to configure instance metadata http tokens for EC2 instances. - by default enforce the tokens to use imdsv2

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-aws/values.schema.json [hunk 1]
```diff
@@ -1158,6 +1158,23 @@
...
                         },
+                        "instanceMetadataOptions": {
+                            "type": "object",
+                            "title": "Instance metadata options",
+                            "description": "Instance metadata options for the EC2 instances in the cluster.",
+                            "properties": {
+                                "httpTokens": {
+                                    "type": "string",
+                                    "title": "HTTP tokens",
+                                    "description": "The state of token usage for your instance metadata requests. If you set this parameter to optional, you can use either the default IAM role or a session policy to control the token usage. If you set this parameter to required, you must use a session policy to control the token usage. Learn more at https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/ .",
```
Comment (AndiDog) [inline helm/cluster-aws/values.schema.json:1169]:
```suggestion
                                    "description": "The state of token usage for your instance metadata requests. If you set this parameter to `optional`, you can use either the default IAM role or a session policy to control the token usage. If you set this parameter to `required`, you must use a session policy to control the token usage. Learn more at [What’s new in IMDSv2](https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/).",
```

Comment (AndiDog) [inline helm/cluster-aws/values.schema.json:1169]: > you must use a session policy to control the token usage

Is it really more than "just" using up-to-date AWS SDKs? From the linked docs, I understood that the restriction to (i.e. checking for) `ec2:RoleDelivery=2.0` isn't really necessary, but it's an optional, additional security measure. If that's true, I'd leave out this complexity from _our_ description here.

And can we mention IMDSv2 more prominently in this description somehow, plus our chosen default?

**Review by bdehri — APPROVED** [review]

**Pull request merged**